pkg/scale/encode.go

```diff
@@ -284,22 +378,46 @@ func (es *encodeState) encodeFixedWidthInt(i interface{}) (err error) {
 	switch i := i.(type) {
 	case int8:
 		err = binary.Write(es, binary.LittleEndian, byte(i))
+		if err != nil {
+			err = fmt.Errorf("encoding FixedWidthInt: %w", err)
```

Comment:
We should not wrap the error with `encoding FixedWidthInt:` here, the caller of `encodeFixedWidthInt` should do it (already does here). Instead you can wrap each with the `i` type. Here it would be:

```suggestion
			err = fmt.Errorf("encoding int8: %w", err)
```

same for below cases